instrumentation/kafka/kafka-clients/kafka-clients-common/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/OpenTelemetryKafkaMetrics.java

```diff
@@ -0,0 +1,225 @@
+ * //    config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, ...);
+ * //    ...
+ * //    try (KafkaProducer<byte[], byte[]> producer = new KafkaProducer<>(config)) { ... }
+ * }</pre>
+ */
+public class OpenTelemetryKafkaMetrics implements MetricsReporter {
+
+  private static final Logger logger = Logger.getLogger(OpenTelemetryKafkaMetrics.class.getName());
+  private static final Set<KafkaMetricId> seenMetrics = ConcurrentHashMap.newKeySet();
+
+  @Nullable private static volatile Meter meter;
+
+  private static final Map<RegisteredInstrument, AutoCloseable> instrumentMap =
+      new ConcurrentHashMap<>();
+
+  /**
+   * Setup OpenTelemetry. This should be called as early in the application lifecycle as possible.
+   * Kafka metrics that are observed before this is called may not be bridged.
+   */
+  public static void setOpenTelemetry(OpenTelemetry openTelemetry) {
+    meter = openTelemetry.getMeter("io.opentelemetry.kafka-clients");
+  }
+
+  /**
+   * Reset for test by reseting the {@link #meter} to {@code null} and closing all registered
+   * instruments.
+   */
+  static void resetForTest() {
+    meter = null;
+    for (Iterator<Map.Entry<RegisteredInstrument, AutoCloseable>> it =
+            instrumentMap.entrySet().iterator();
+        it.hasNext(); ) {
+      try {
+        it.next().getValue().close();
+        it.remove();
+      } catch (Exception e) {
+        throw new IllegalStateException("Error occurred closing instrument", e);
+      }
+    }
+  }
+
+  /**
+   * Print a table mapping kafka metrics to equivalent OpenTelemetry metrics, in markdown format.
+   */
+  static void printMappingTable() {
+    StringBuilder sb = new StringBuilder();
+    // Append table headers
+    sb.append(
+            "| # | Metric Group | Metric Name | Attribute Keys | Instrument Name | Instrument Description | Instrument Type |")
+        .append(lineSeparator())
+        .append(
+            "|---|--------------|-------------|----------------|-----------------|------------------------|-----------------|")
+        .append(lineSeparator());
+    Map<String, List<KafkaMetricId>> kafkaMetricsByGroup =
+        seenMetrics.stream().collect(groupingBy(KafkaMetricId::getGroup));
+    int count = 1;
+    // Iterate through groups in alpha order
+    for (String group : kafkaMetricsByGroup.keySet().stream().sorted().collect(toList())) {
+      List<KafkaMetricId> kafkaMetricIds =
+          kafkaMetricsByGroup.get(group).stream()
+              .sorted(comparing(KafkaMetricId::getName))
+              .collect(toList());
+      // Iterate through metrics in alpha order by name
+      for (KafkaMetricId kafkaMetricId : kafkaMetricIds) {
+        Optional<InstrumentDescriptor> descriptor =
+            Optional.ofNullable(KafkaMetricRegistry.getInstrumentDescriptor(kafkaMetricId));
+        // Append table row
+        sb.append(
+            String.format(
+                "| %s | %s | %s | %s | %s | %s | %s |%n",
```

Comment:
That's totally minor, but WDYT about applying some more markdown formatting to the table?
```suggestion
                "| %s | `%s` | `%s` | %s | `%s` | %s | `%s` |%n",
```